extensions/wikia/BrokenRenameFix/SpecialBrokenRenameFixController.class.php

```diff
@@ -0,0 +1,166 @@
+			$task->call( 'rerunRenameScript', $this->userId, $this->oldName, $this->newName );
+			$task->prioritize();
+			$task->queue();
+
+			$this->addNotice( wfMessage( 'brf-success' )->escaped() . $this->getLogsLink(), 'success' );
+		} else {
+			$this->prefillFormValues();
+		}
+	}
+
+	/**
+	 * Validates the form's input and returns a status of the checks. It also adds notices
+	 * on each step if something goes wrong.
+	 *
+	 * @return bool
+	 */
+	private function validateForm() {
+		$status = true;
+
+		$editToken = $this->request->getVal( 'brf-form-token' );
+		if ( !$this->request->wasPosted() || !$this->wg->User->matchEditToken( $editToken ) ) {
+			$this->addNotice( wfMessage( 'brf-error-invalid-request' )->escaped() );
+			$status = false;
+		}
+
+		$this->userId = $this->request->getVal( 'brf-form-user-id' );
+		$this->oldName = $this->request->getVal( 'brf-form-old-name' );
+		$this->newName = $this->request->getVal( 'brf-form-new-name' );
+
+		if ( empty( $this->userId ) || empty( $this->oldName ) || empty( $this->newName ) ) {
+			$this->addNotice( wfMessage( 'brf-error-empty-fields' )->escaped() );
+			$status = false;
+		}
+
+		$usernameFromId = User::newFromId( $this->userId )->getName();
+		if ( $usernameFromId !== $this->oldName && $usernameFromId !== $this->newName ) {
+			$this->addNotice( wfMessage( 'brf-error-invalid-user' )->escaped() );
+			$status = false;
+		}
+
+		return $status;
+	}
+
+	/**
+	 * Sets values used to prefill form if it was submited with errors.
+	 */
+	private function prefillFormValues() {
+		$this->userIdValue = $this->userId;
+		$this->oldNameValue = $this->oldName;
+		$this->newNameValue = $this->newName;
+	}
+
+	/**
+	 * Sets edit token and action URL in the form.
+	 */
+	private function setFormData() {
+		$this->editToken = $this->wg->User->getEditToken();
+		$this->actionUrl = $this->specialPage->getTitle()->getLocalURL( [
+			self::PAR => self::BRF_FORM_SUBMIT
+		] );
+	}
+
+	/**
+	 * Sets all labels and description messages.
+	 */
+	private function setMessages() {
+		$this->description = wfMessage( 'brf-desc' )->escaped();
+		$this->userIdLabel = wfMessage( 'brf-label-user-id' )->escaped();
+		$this->oldNameLabel = wfMessage( 'brf-label-old-name' )->escaped();
+		$this->newNameLabel = wfMessage( 'brf-label-new-name' )->escaped();
+		$this->submitText = wfMessage( 'brf-label-submit' )->escaped();
+	}
+
+	/**
+	 * Adds a notice that are displayed above the form.
+	 *
+	 * @param $text
+	 * @param string $type 'error' or 'success'
+	 */
+	private function addNotice( $text, $type = 'error' ) {
+		$this->notices[] = Html::rawElement( 'p',
+			[
+				'class' => sprintf( 'brf-form-notice-%s', $type ),
+			],
+			$text
+		);
+	}
+
+	/**
+	 * Creates a link to logs that will be available after the rename rerun finishes.
+	 *
+	 * @return string An HTML <a> element
+	 * @throws MWException
+	 */
+	private function getLogsLink() {
+		$logTitleText = sprintf( '%s/%s to %s rename rerun log',
```

Comment:
This isn't used any more, right? i.e. we're not saving the log on the wikia, so this link can be removed.